src-ghc/Pact/ApiReq.hs

```diff
@@ -404,12 +425,12 @@ withKeypairsOrSigner unsignedReq ApiReq{..} keypairAction signerAction =
     toSigner ApiSigner{..} = Signer _asScheme _asPublic _asAddress (fromMaybe [] _asCaps)
 
 
-mkApiReqExec :: Bool -> ApiReq -> FilePath -> IO ((ApiReq,String,Value,PublicMeta),Command Text)
+mkApiReqExec :: Bool -> ApiReq -> FilePath -> IO (ApiReqParts,Command Text)
 mkApiReqExec unsignedReq ar@ApiReq{..} fp = do
   (code,cdata) <- withCurrentDirectory (takeDirectory fp) $ do
     code <- case (_ylCodeFile,_ylCode) of
       (Nothing,Just c) -> return c
-      (Just f,Nothing) -> liftIO (readFile f)
+      (Just f,Nothing) -> liftIO (T.readFile f)
```

Comment:
The general consensus on `readFile` for small static data is to stick with `ByteString`'s `readFile` and encoding. It's both faster, and less memory intensive. `Text`'s `readFile` is both suboptimal and prone to data leaks.